realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -211,7 +211,8 @@ static jlong findAllMultiSortedWithHandover
                 case type_Double:
                 case type_String:
                 case type_Timestamp:
-                    indices.push_back( S(long_arr[i]) );
+                    field_ref[0] = S(long_arr[i]); // TODO Enable support for sorting through links
+                    indices.push_back( field_ref );
```

Comment:
`indices.push_back(std::vector<size_t> {S(long_arr[i])})` should just work. To avoid calling the copy constructors. Same as below.